chain/neutrino.go

```diff
@@ -358,6 +358,11 @@ func (s *NeutrinoClient) Rescan(startHash *chainhash.Hash, addrs []btcutil.Addre
 	// notification indicating the rescan has "finished".
 	if header.BlockHash() == *startHash {
 		s.finished = true
+
+		// Release the lock while dispatching the notification since
+		// it's possible for the notificationHandler to be waiting to
+		// acquire it before receiving the notification.
+		s.clientMtx.Unlock()
```

Comment:
Nice, I think this is an ok solution for now, this should fix the issue I ran into. Longer term we should move to a more granular set of mutexes (or get rid of the mutexes all together), but short term this is fine :+1: